### DISCUSSION TITLE
Map gestures in navigation

### DISCUSSION THREAD

**Discussion by Maverikk65** [devemux86/cruiser#678] [Q&A] [answered]
Hello, 
while driving on the weekend my co-driver wanted to zoom the map with the fingers. Just as we are used with "every other" navigation. This seems not to work. Only the +/- dots seem to work. Or switching off navigation and then zoom with fingers... 

It's a bug? It's a feature? And if so --- why? 

Tnx. Danke. VG, NG.

**Comment by devemux86** [comment]
@Maverikk65 Map gestures are not enabled in navigation.
Can you do map gestures while driving or with gloves?

Zoom buttons are more suitable for this.
Or use the volume keys, if enabled in "Settings | Map".

**Reply by Maverikk65** [reply]
Hello Emux, many thanks for the reply. So, it is a feature ;-) 

I have been thinking about it a short while now --- and THE question is: The "gloves" argument is pointing at motobike drivers... Also mountainbike or street bike. But. Not at pedetrians (except in cold winter days, and then the gloves might not be suitable to actually work on the screen), maybe on runners or wanderers, but surely not for car drivers... Am I right?! 

So, the target group you are aiming for is not totally clear to me... For my person I can only say that I very frequently use(d) this function on other navigations (to zoom with two fingers and adjust the view where to go accordingly) and I really love this feature, in Cruiser it is much more dificult at the moment (zoom by the buttons and then drag the map to the desired view). 

BTW: I am working myself into your app now. It is mighty, very much customable which I already love a lot! The wiki is missing many pages however. I would love to make some screenshots and wiki pages while working me down and in, but I surely have quite some questions while doing so... (I use to write little manuals for software applications sometimes). Just as an offer to contribute to the project...
Danke, VG, NG.

**Reply by devemux86** [reply]
Cruiser welcomes all travel modes: car, motorcycle, bike, hiking, running, walking, etc.
Just check the available app profiles and the different routing options per profile.

Map gestures in navigation can be enabled in a next update to see how it works.

---

@Maverikk65 About the Wiki it is open for everyone to contribute and write to it.
If you have the time and want to write to the Wiki you are certainly welcome! :+1:

**Reply by 0709wiwiwi** [reply]
@Maverikk65 

"I am working myself into your app now." 

[You may consult some how to items](https://github.com/devemux86/cruiser/wiki/How-to%3F)
- How to display usefull Cruiser app settings hints.

**Reply by 0709wiwiwi** [reply]
@Maverikk65  Something for the wiki ?

Correct Misplaced shaping_via waypoints tool.
The usefulness may not always be clear ?

In BE/NE cycling junction (poi) routing is very popular.
The cycle junctions (17456) can be downloaded as favorites. 
By the favorite poi's you can then click together a cycling route. 
The poi positions can differ slightly from the direction to follow.  

Example.
- BRouter offline with correction.
[Brouter_offline.gpx.txt](https://github.com/user-attachments/files/18020764/Brouter_offline.gpx.txt)

- GH online without correction.
[GH_online.gpx.txt](https://github.com/user-attachments/files/18020768/GH_online.gpx.txt)

Fietsknooppunten Y2024.
[knooppunten_2024.gpx.txt](https://github.com/user-attachments/files/18020794/knooppunten_2024.gpx.txt)

**Reply by Maverikk65** [reply]
Great, many thanks, I will look on those over the weekend when I find some time... 
Danke, VG, NG.

**Comment by DolomitesTom** [comment]
> Map gestures in navigation can be enabled in a next update to see how it works.

👍

**Comment by devemux86** [comment]
Map gestures in navigation are available in **[Cruiser 5.1.0-b14](https://github.com/devemux86/cruiser/discussions/671)**.

**Reply by Maverikk65** [reply]
Hello Emux @devemux86 , 
sorry I am a bit late... I wanted to thank you for the massive improvements (at least for me) and tell about my driving experience this Friday... 
I had to travel from work in Bremerhaven to my hometown of Gotha, about 400 km and 4 hours, mostly autobahn and some country road... I normally start at around 1700-1800h because of the Friday trafic situation. Before I drive such a tour I take a look at www.verkehrslage.de, which paid off very well --- around Hildesheim there is a looong construction site (15 km) in both directions and it was blocked, showing a passing time of 238 minutes... 
https://www.hildesheimer-allgemeine.de/meldung/achtung-die-autobahn-in-richtung-sueden-ist-derzeit-voll-gesperrt.html

So I planned my route on the pc with www.kurviger.de as I normally do in the last years... However with a glimpse on Google maps I saw several road blocks which I avoided. Finally I took off at 1900h only and actually planned Hannover A2 to Hämelerwald and then South on the country roads --- if you take the "normal detour" you will be standing in a queue with 1000s other cars and trucks. 

It's very nice to use the .kurviger import format, as I think there will be no re-routing on such a tour, as it alsways happens with other devices (like my former Copilot). A bit annoying was the fact, that Google drive is blocked on the company network, so I had to save in Kurviger cloud, open on the tablet, save in Google drive again and then import into Cruiser. But it's okay... 

So I was driving with 3 navigation devices: Cruiser, I ran Google Maps on the smartphone, and the internal navigation (I believe it's a TomTom...). However, although the other two recognized the trafic situation (live trafic) they wanted to re-route me very near to the autobahn, which is just useless... So I even had to switch them off and concentrated on Cruiser ;-) The zoom with the lower factor (set to 9) is very nice for me, although it used to be a bit smoother in Copilot (and there was that function "zoom on next waypoint, which I just loved). But driving on the country road which was unknown to me, in the night, was very good! Several time I did a quick glimpse on the overview and zoomed out and in with the map gestures --- very good! Many thanks for this. Also that it does NOT rotate, this is very good! 

So I really enjoyed the new features A LOT (map gestures, zoom to lower level). Many thanks for that wonderful work. I said it before at another thread --- your night map is very good for orientation! 
CU, Danke, VG, NG.

**Reply by devemux86** [reply]
@Maverikk65 Thanks for the detailed report! :+1: 

> there was that function "zoom on next waypoint"

Can you describe how it works?

**Reply by Maverikk65** [reply]
Hello Emux, 
many thanks for your interest ;-) And sorry for late reply again... I will try to describe. 

Zoom to next waypoint was one option in Copilot navigation. This means, the zoom factor was VERY dynamic. When I started a tour in Berlin and drive for 200km straight on the autobahn the zoom was VERY large and for about one hour you did not really notice that you were driving... However, the nearer you came to your first waypoint the more detailed was the zoom, unti about 300 meters, when the next waypoint was calculated and the map zoomed out again. Until about 40 km before that next waypoint the map was pointing northwards, after that in the direction of movement. 

So, when driving a tour as you normally have to set a shaping point 1-2 km after each crossing you had always a nice view of the surrounding landscape and also of the directions where to go, as it would zoom into the junction as well. This zooming worked very smooth. 

When I drove with Cruiser last Friday I had zoom setting min = 9 and max = 17. With this setting when I drove on the A7 into the Autobahndreieck Hannover-Nord at about 160km/h all of a sudden the Cruiser app zoomed to max which was a bit "unsmooth" ;-) BUT --- I think the zoom settings in Cruiser are very good now and work fine! So you shouldn't care about it, you might rather "verschlimmbessern" (brake) your good product! 

I will drive tomorrow to Frankfurt via the country roads, maybe I will give another feedback ;-) 

Ciao, VG, NG.

**Reply by devemux86** [reply]
@Maverikk65 

There are various auto zoom modes in Cruiser (enable "Extended settings"):
- **Speed**: auto zoom is dynamic and based on the current speed
(adjusted by min / max zoom and min / max speed settings)

- **Distance**: auto zoom is based on the distance to the next turn instruction
(adjusted by min / max zoom settings)

**Reply by Maverikk65** [reply]
Yes, thanks! I have not been drving enough to see and find out the full potential of this and which settings might please me the most...

**Comment by devemux86** [comment, marked as answer]
- Map gestures in navigation is available in https://github.com/devemux86/cruiser/discussions/684
